### PR TITLE
CA-195440: Mark VM stuck only once balloon-driver has published featur…

### PR DESCRIPTION
**Pull request by koushikcgit**
…e-balloon

Signed-off-by: Koushik Chakravarty <koushik.chakravarty@citrix.com>

### CODE REVIEW

File: src/squeeze_xen.ml [hunk 1]
```diff
@@ -370,9 +375,10 @@ module Domain = struct
 end
 
 (** Mark the domain as squeezed declared stuck *)
-let declare_domain_stuck ~xc domid =
+let declare_domain_stuck ~xc domid value =
```
Comment (robhoes) [inline src/squeeze_xen.ml]: Do we ever pass in a `value` other than `"true"` now?

File: src/squeeze_xen.ml [hunk 2]
```diff
@@ -370,9 +375,10 @@ module Domain = struct
...
 	let cnx = xc in
-	Domain.set_domain_stuck_noexn cnx domid "true"
+	let can_balloon = Domain.get_feature_balloon cnx domid in
+	if can_balloon then Domain.set_domain_stuck_noexn cnx domid value
```
Comment (robhoes) [inline src/squeeze_xen.ml]: So the `declare_domain_stuck` function now does not always mark a domain as "stuck". Would it make more sense to have to conditional at the point where `declare_domain_stuck` is called? Or rename the function.

**Comment by robhoes** [comment]
In the commit message: "ballon" -> "balloon".

**Comment by robhoes** [comment]
I think the patch is good in principle. Just a few small issues.

File: src/squeeze_xen.ml (hunk 1 above)
Comment (koushikcgit) [inline src/squeeze_xen.ml]: Not really - Just added the flexibility to add it in future.
I will make changes to rename the function

**Pull request merged**